corehq/apps/accounting/invoicing.py

```diff
@@ -161,9 +162,17 @@ def _get_invoice_end(sub, date_end):
         if record.should_send_email:
             try:
                 if invoice.subscription.service_type == SubscriptionType.IMPLEMENTATION:
-                    record.send_email(contact_email=invoice.account.dimagi_contact, cc_emails=self.recipients)
+                    if self.recipients:
+                        for email in self.recipients:
+                            record.send_email(contact_email=email)
+                    elif invoice.account.dimagi_contact:
+                        record.send_email(contact_email=invoice.account.dimagi_contact,
+                                          cc_emails=[settings.ACCOUNTS_EMAIL])
+                    else:
+                        record.send_email(contact_email=settings.ACCOUNTS_EMAIL)
                 else:
-                    for email in self.recipients:
+                    for email in self.recipients or \
+                                 record.invoice.subscription.account.billingcontactinfo.email_list or []:
```

Comment:
This should use ```record.invoice.contact_emails``` to handle the situation when ```billing_contact_info``` is nonexistent or ```billing_contact_info.email_list``` is empty.  See https://github.com/dimagi/commcare-hq/blob/9911b02bc3b3147cb7d7873312acce20a218e0d1/corehq/apps/accounting/models.py#L1802-L1823

What do you think about adding an extra test for that case?

Reply:
It looks like ```test_product_subscription``` is a copy of ```test_unspecified_recipients_product```.  The new case to consider is when ```billing_contact_info.email_list = []``` and there are WebUsers in the project space.  Does it make sense why that's a separate test case?